ui/brave_custom_notification/message_view.cc

```diff
@@ -0,0 +1,301 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ui/brave_custom_notification/message_popup_view.h"
+#include "brave/ui/brave_custom_notification/message_view.h"
+#include "brave/ui/brave_custom_notification/notification_background_painter.h"
+#include "brave/ui/brave_custom_notification/notification_control_buttons_view.h"
+#include "brave/ui/brave_custom_notification/public/cpp/constants.h"
+#include "base/strings/utf_string_conversions.h"
+#include "build/build_config.h"
+#include "ui/accessibility/ax_enums.mojom.h"
+#include "ui/accessibility/ax_node_data.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/compositor/scoped_layer_animation_settings.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/color_palette.h"
+#include "ui/gfx/image/image_skia_operations.h"
+#include "ui/gfx/shadow_util.h"
+#include "ui/gfx/shadow_value.h"
+#include "ui/strings/grit/ui_strings.h"
+#include "ui/views/background.h"
+#include "ui/views/border.h"
+#include "ui/views/controls/button/image_button.h"
+#include "ui/views/controls/highlight_path_generator.h"
+#include "ui/views/controls/image_view.h"
+#include "ui/views/controls/scroll_view.h"
+#include "ui/views/focus/focus_manager.h"
+#include "ui/views/style/platform_style.h"
+#include "ui/views/widget/widget.h"
+
+#if defined(OS_WIN)
+#include "ui/base/win/shell.h"
+#endif
+
+namespace brave_custom_notification {
+
+namespace {
+bool ShouldShowAeroShadowBorder() {
+#if defined(OS_WIN)
+  return ui::win::IsAeroGlassEnabled();
+#else
+  return false;
+#endif
+}
+
+}  // namespace
+
+// static
+// const char kViewClassName[] = "MessageView";
+
+class MessageView::HighlightPathGenerator
+    : public views::HighlightPathGenerator {
+ public:
+  HighlightPathGenerator() = default;
+
+  SkPath GetHighlightPath(const views::View* view) override {
+    return static_cast<const MessageView*>(view)->GetHighlightPath();
+  }
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(HighlightPathGenerator);
+};
+
+MessageView::MessageView(const Notification& notification) : notification_id_(notification.id()), slide_out_controller_(this, this) {
+  SetFocusBehavior(FocusBehavior::ALWAYS);
+  views::HighlightPathGenerator::Install(
+      this, std::make_unique<HighlightPathGenerator>());
+
+  // Paint to a dedicated layer to make the layer non-opaque.
+  SetPaintToLayer();
+  layer()->SetFillsBoundsOpaquely(false);
+
+  UpdateWithNotification(notification);
+
+  UpdateCornerRadius(0, 0);
+
+  // If Aero is enabled, set shadow border.
+  if (ShouldShowAeroShadowBorder()) {
+    const auto& shadow = gfx::ShadowDetails::Get(2, 0);
```

Comment:
Can we please move hardcoded values to `const` inside an empty namespace in the `.cc` file

Reply:
Would prefer not to change this line, as this code was copy and pasted